src/lib.rs

```diff
@@ -152,6 +160,9 @@ pub mod pallet {
                 GasStrategy::Metered(gas_amount),
             );
 
+            // Apply true transactions on blockchain.
+            balance.apply_transactions()?;
+
```

Comment:
We'll need to write a test with a Move script that:
1. executes the transfer successfully within the script
2. then the script fails due to some other failure

That test should make sure the `result::from_vm_result` on the line below will return an error.

We should make sure the balances are not updated in that Rust test. (I'm not sure if that's the case with the current implementation)

Reply:
Do you want to add this now or in the next PR?

Reply:
In the next PR is fine